sklearn/linear_model/logistic.py

```diff
@@ -1538,6 +1538,15 @@ def fit(self, X, y, sample_weight=None):
             raise ValueError("class_weight provided should be a "
                              "dict or 'balanced'")
 
+        # compute the class weights for the entire dataset y
+        if self.class_weight in ("auto", "balanced"):
```

Comment:
why do we need this here?  logistic_regression_path later on should do that, right?